src/Hl7.Fhir.Specification/Specification/Snapshot/SnapshotGenerator.cs

```diff
@@ -1167,6 +1167,9 @@ private async T.Task<bool> mergeTypeProfiles(ElementDefinitionNavigator snap, El
                             // Rebase before merging
                             var rebasedRootElem = (ElementDefinition)typeRootElem.DeepCopy();
                             rebasedRootElem.Path = diff.Path;
+                            // MV 20210727: copy cardinality from base (so do not use the cardinality of the type). See issue #1824
```

Comment:
Doesn't this mean that if you have an extension definition (which is a profiled Extension), that it will ignore the extension's cardinality now too?  Isn't it safer (for regressions) to take the higher one of the "min" and the lower one of the "max", so that you can still override this cardinality in a profile (at least for Extension)?

Reply:
So, I had a lot of thinking about this and I was experimenting with Extension as well and I could not find an error there.

I think the cardinality of a type (what does that actually mean?)  should influence the cardinality of an element. The cardinality of `Observation.code` is `1..1`. When you profile this and you change the type of `Observation.code` to a profiled `CodeableConcept`  with cardinality `2..*` (again, what does that mean?), then the cardinality of the element `Observation.code` should still be `1..1` to my opinion.